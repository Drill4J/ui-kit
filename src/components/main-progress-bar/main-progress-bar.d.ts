interface Props {
  className?: string;
  value: string;
  type?: 'primary' | 'secondary';
  testContext?: string;
}
export declare const MainProgressBar: {
  ({
    className, value, type, testContext,
  }: Props): JSX.Element;
  displayName: string;
  __docgenInfo: {
    description: string;
    displayName: string;
    props: {
      className: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
      value: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
      type: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
      testContext: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
    };
  };
};
export {};
