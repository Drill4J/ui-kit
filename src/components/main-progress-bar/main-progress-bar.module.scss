@import '../../style-guide/index';

.main-progress-bar {
  height: 32px;
  border-radius: 4px;
  background: map-get($monochrome, light-tint);

  &__progress {
    height: 32px;
    border-radius: 4px;
    background: map-get($data-visualization, build-cover);
    transition: width 2000ms ease-in;

    &--type {
      &--primary {
        background: map-get($data-visualization, scope-cover);
      }

      &--secondary {
        background: map-get($data-visualization, overlapping);
      }
    }
  }
}
