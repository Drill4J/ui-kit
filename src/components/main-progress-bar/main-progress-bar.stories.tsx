import React from 'react';
import { storiesOf } from '@storybook/react';

import { MainProgressBar } from './main-progress-bar';

storiesOf('MainProgressBar', module)
  .add('MainProgressBar with change coverage panel', () => {
    const [coverage, setCoverage] = React.useState(20);
    const handleChange = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
      setCoverage(Number(value));
    };
    return (
      <div style={{ display: 'grid', gap: '20px', padding: '40px' }}>
        <MainProgressBar value={`${coverage}%`} />
        <MainProgressBar value={`${coverage}%`} type="primary" />
        <MainProgressBar value={`${coverage}%`} type="secondary" />
        <input type="range" min="0" max="100" value={coverage} onChange={handleChange} />
      </div>
    );
  });
