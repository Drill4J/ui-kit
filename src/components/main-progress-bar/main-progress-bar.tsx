import * as React from 'react';
import { BEM } from '@redneckz/react-bem-helper';

import styles from './main-progress-bar.module.scss';

interface Props {
  className?: string;
  value: string;
  type?: 'primary' | 'secondary';
  testContext?: string;
}

const mainProgressBar = BEM(styles);

export const MainProgressBar = mainProgressBar(({
  className, value, type, testContext,
}: Props) => (
  <div className={className} data-test={`main-progress-bar:${(testContext || type)}`}>
    <Progress style={{ width: value }} type={type} />
  </div>
));

const Progress = mainProgressBar.progress('div');
