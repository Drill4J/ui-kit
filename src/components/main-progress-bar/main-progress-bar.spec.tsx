import React from 'react';
import renderer from 'react-test-renderer';
import { MainProgressBar } from './main-progress-bar';

describe('AdditionalProgressBar', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<MainProgressBar value="50%" type="primary" testContext="snapshot" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
