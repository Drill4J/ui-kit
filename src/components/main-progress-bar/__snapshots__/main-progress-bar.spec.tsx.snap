// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalProgressBar should match snapshot 1`] = `
<div
  className="main-progress-bar___StyledDiv-sc-s1ed9u-1 lIbBF"
  data-test="main-progress-bar:snapshot"
>
  <div
    className="main-progress-bar__Progress-sc-s1ed9u-0 gxVmIi"
    style={
      Object {
        "width": "50%",
      }
    }
    type="primary"
  />
</div>
`;
